### PR TITLE
feat: adds Table

### PR DESCRIPTION
**Pull request by aditya-kumawat**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits
#43 
...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: css/src/components/table.css [hunk 1]
```diff
@@ -1,143 +1,337 @@
...
 
-.loader-row:not(:last-child) {
-    border-bottom: 1px solid #efefef;
+.Table--pinned {
+    position: absolute;
+    height: 100%;
+    z-index: 2;
+    overflow-x: hidden;
+    /* overflow: hidden; */
+    -ms-overflow-style: none;
```
Comment (atifzaidi92) [inline css/src/components/table.css:35]: I think we are using postcss, we can manage this prop there

File: css/src/components/table.css [hunk 2]
```diff
@@ -1,143 +1,337 @@
...
+    border-right: 3px solid var(--secondary);
 }
 
-.loader-column {
-    padding: 4px 8px;
-    text-align: left;
+.Table-pinned::-webkit-scrollbar {
```
Comment (atifzaidi92) [inline css/src/components/table.css:39]: what about other browsers

File: css/src/components/table.css [hunk 3]
```diff
@@ -1,143 +1,337 @@
...
+    display: none;
+}
+
+/* .Table--pinned .Table-cell {
```
Comment (atifzaidi92) [inline css/src/components/table.css:43]: remove this

File: css/src/components/table.css [hunk 4]
```diff
@@ -1,143 +1,337 @@
...
+    top: 0;
+    z-index: 1;
+}
+
+.Table-headerCell {
+    display: flex;
+    height: 100%;
+    flex-direction: row;
     align-items: center;
+    /* padding-left: var(--spacing-l); */
```
Comment (atifzaidi92) [inline css/src/components/table.css:95]: remove this

File: css/src/components/table.css [hunk 5]
```diff
@@ -1,143 +1,337 @@
...
-    overflow-y: hidden;
+.Table .Dropdown {
+    min-width: unset;
 }
 
-.TableGrid-scroll .TableGrid-scroll--center::-webkit-scrollbar {
-    -webkit-appearance: none;
-    height: 10px;
+.Popover {
+    width: 200px !important;
```
Comment (atifzaidi92) [inline css/src/components/table.css]: design need?

File: core/components/organisms/table/Table.tsx [hunk 6]
```diff
@@ -1,107 +1,1145 @@
...
-import { Props, TableState } from './interfaces';
+import classNames from 'classnames';
+import { Button, Checkbox, Dropdown, Icon, Heading, Pagination, Placeholder, PlaceholderParagraph } from '@/index';
+import { StatusHintsProps, IconProps, InputProps, DropdownProps, PaginationProps } from '@/index.type';
+import TableCell, { TableCellProps } from './TableCell';
 
-export interface TableProps extends Props {
+export type SortType = 'asc' | 'desc';
+export type Alignment = 'left' | 'right' | 'center';
+export type Comparator = (a: Data, b: Data) => number;
```
[the remark this answers is not in the document]
Comment (aditya-kumawat) [reply core/components/organisms/table/Table.tsx]: Will update this.

File: core/components/organisms/table/Table.tsx [hunk 7]
```diff
@@ -1,107 +1,1145 @@
...
+export type fetchDataFn = (attr: FetchDataAttr) => Promise<{
+  totalRecords: number,
+  data: Data[],
+  schema: CellSchema[]
+}>;
+export type updateSelectAllFn = (attr: TableState['selectAll']) => void;
+export type updateCellSchemaFn = (name: string, schemaUpdate: Partial<StateSchema>) => void;
+export type updateRowDataFn = (rowIndexes: number[], dataUpdate: Partial<StateData>) => void;
+export type updateReorderHighlighterFn = (dim: number | null) => void;
+export type sortDataFn = (comparator: Comparator, type: SortType) => void;
```
[the remark this answers is not in the document]
Comment (aditya-kumawat) [reply core/components/organisms/table/Table.tsx]: Need to discuss with Design team.

**Pull request merged**